### ISSUE TITLE
[Bug] Add/Remove component and Destroy entity cause LastChunk error

### ISSUE THREAD

**Issue by Sarofc** [genaray/Arch#173]
Reproduce: add DestroyForloop to WorldTest.cs 

Result: System.IndexOutOfRangeException : Index was outside the bounds of the array.

    [Test]
    public void DestroyForloop()
    {
        var _entitiesToChangeColor = new QueryDescription().WithAll<Transform>();

        var entities = new List<Entity>();
        const int len = 1000;
        for (int i = 0; i < len; i++)
        {
            var ent = _world.Create(_entityGroup);
            entities.Add(ent);
        }

        for (int i = 8; i < entities.Count; i++)
        {
            var ent = entities[i];
            if (i % 3 == 0)
            {
                // A demonstration of bulk adding and removing components.
                _world.Add(in _entitiesToChangeColor, 1);
                _world.Remove<int>(in _entitiesToChangeColor);

                _world.Destroy(ent);
            }
        }
    }

**Comment by genaray** [comment]
Thanks! Ill look into this soon :)

**Comment by genaray** [comment]
Fixed, lemme know if it worked! :) 
Not released as a nugget yet, but on the master haha

**Comment by genaray** [comment]
No its not fixed yet, dammit. My mistake

**Comment by Mathetis** [comment]
I'm seeing what might be a similar crash on 1.2.7 alpha. When adding, removing components and destroying and creating new entities, I get an out of index exception. Looking at the debugger, I see this, with chunk capacity and size at 250.

Adding an empty component (a struct with no params) causes an IndexOutsideOfBoundsException

  | Name | Value | Type
-- | -- | -- | --
◢ | Chunk | {Chunk = { Capacity = 250, Size = 250 }} | Arch.Core.Chunk
  | Capacity | 250 | int
  | ▶ ComponentIdToArrayIndex | {int[20]} | int[]
  | ▶ Components | {System.Array[7]} | System.Array[]
  | ▶ Entities | {Arch.Core.Entity[250]} | Arch.Core.Entity[]
  | Size | 250 | int
▶ | Components | {object[7]} | object[]
▶ | EntityInfo | {Arch.Core.EntityInfo} | Arch.Core.EntityInfo
  | Id | 2026 | int
  | IsAlive | true | bool
  | Version | 1 | int
▶ | World | {World { Id = 0, Capacity = 15182, Size = 1672 }} | Arch.Core.World



</body>
</HTML>

**Comment by Mathetis** [comment]
I think something else is going on in my case. I'll work on opening another bug.

I noticed that the byte size of the types was an even number before the archetype change, and an odd number after addition.

The entity that experiences the IndexOutsideOfBounds has 59 bytes in the archetype. This is because empty structs take 1 byte.

Originally, the entity had 7 components totaling 58 bytes. Adding one more 1 byte component would throw an index out of range sometimes.
I padded some structs to 2 byte. Ironically I missed one. But now it is 59 bytes before, and adding a 2 byte component makes it 61 bytes. It no longer throws an index outside of range exception, or at least I cannot reproduce it now.

Prior:
![image](https://github.com/genaray/Arch/assets/11789696/bf5f9aea-9ef7-4bb7-be20-01de4550ff19)

After adding:
![image](https://github.com/genaray/Arch/assets/11789696/d6077170-fbcc-431e-bf98-2185e9197c4d)

I'll keep looking at this as I think this is a different bug than this one.
